project/CrossJava.scala

```diff
@@ -0,0 +1,135 @@
+/**
+  * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+  */
+
+package akka
+
+import java.io.File
+
+import sbt._
+
+import scala.collection.immutable.ListMap
+
+case class JavaVersion(numbers: Vector[Long], vendor: Option[String]) {
+  def numberStr: String = numbers.mkString(".")
+
+  def withVendor(vendor: Option[String]) = copy(vendor = vendor)
+  def withVendor(vendor: String) = copy(vendor = Option(vendor))
+  def withNumbers(numbers: Vector[Long]) = copy(numbers = numbers)
+
+  override def toString: String = {
+    vendor.map(_ + "@").getOrElse("") + numberStr
+  }
+}
+object JavaVersion {
+  def apply(version: String): JavaVersion = CrossJava.parseJavaVersion(version)
+  def apply(numbers: Vector[Long], vendor: String): JavaVersion = new JavaVersion(numbers, Option(vendor))
+}
+
+
+/** Tools for discovering different Java versions,
+  * will be in sbt 1.1.7 (https://github.com/sbt/sbt/pull/4139) but until that time replicated here */
+object CrossJava {
+  object Keys {
+    val discoveredJavaHomes = settingKey[Map[String, File]]("Discovered Java home directories")
+    val javaHomes = settingKey[Map[String, File]]("The user-defined additional Java home directories")
+    val fullJavaHomes = settingKey[Map[String, File]]("Combines discoveredJavaHomes and custom javaHomes.")
+  }
+
+  import Keys._
+
+  val crossJavaSettings = Seq(
+    discoveredJavaHomes := CrossJava.discoverJavaHomes,
+    javaHomes := ListMap.empty,
+    fullJavaHomes := CrossJava.expandJavaHomes(discoveredJavaHomes.value ++ javaHomes.value),
```

Comment:
is the order of importance? should user-defined be first to be able to override?

Reply:
@2m WDYT?

Reply:
Yes, user-defined ones should override the discovered ones.